File: Graphs/BFSGraphs/examples/bfs_graph_example.js
const BFSGraph = require('../source/bfs_graph');

const graph = new BFSGraph();

// Adding edges to the graph
graph.addEdge("A", "B");
graph.addEdge("A", "C");
graph.addEdge("B", "D");
graph.addEdge("C", "D");
graph.addEdge("D", "E");
graph.addEdge("E", "F");
graph.addEdge("C", "F");

// Display the graph structure
console.log("\nGraph Adjacency List:");
graph.display();

// Perform BFS starting from vertex 'A'
console.log("\nBFS Traversal Starting from Vertex A:");
graph.bfs("A");

// Perform BFS starting from vertex 'C'
console.log("\nBFS Traversal Starting from Vertex C:");
graph.bfs("C");

// Perform BFS starting from vertex 'D'
console.log("\nBFS Traversal Starting from Vertex D:");
graph.bfs("D");

// Perform BFS starting from a non-existent vertex 'Z'
console.log("\nBFS Traversal Starting from Vertex Z (Non-Existent):");
graph.bfs("Z");

File: Graphs/BFSGraphs/examples/bfs_graph_example.py
from Graphs.BFSGraphs.source.bfs_graph import BFSGraph


if __name__ == "__main__":
    # Create a graph instance
    graph = BFSGraph()

    # Add edges to the graph
    graph.add_edge("A", "B")
    graph.add_edge("A", "C")
    graph.add_edge("B", "D")
    graph.add_edge("C", "D")
    graph.add_edge("D", "E")
    graph.add_edge("E", "F")
    graph.add_edge("C", "F")

    # Display the graph structure
    print("\nGraph Adjacency List:")
    graph.display()

    # Perform BFS starting from vertex 'A'
    print("\nBFS Traversal Starting from Vertex A:")
    graph.bfs("A")

    # Perform BFS starting from vertex 'C'
    print("\nBFS Traversal Starting from Vertex C:")
    graph.bfs("C")

    # Perform BFS starting from vertex 'D'
    print("\nBFS Traversal Starting from Vertex D:")
    graph.bfs("D")

    # Perform BFS starting from a vertex that does not exist
    print("\nBFS Traversal Starting from Vertex Z (Non-Existent):")
    graph.bfs("Z")
